chain/client/src/client_actor.rs

```diff
@@ -2096,6 +2065,68 @@ impl ClientActorInner {
         }
         true
     }
+
+    /// Produces `num_blocks` number of blocks.
+    ///
+    /// New blocks are placed on top of `prev_block_height`,
+    /// if present, otherwise on top of current chain's head.
+    /// The parameter `block_height` controls the new blocks' own height.
+    #[cfg(feature = "test_features")]
+    pub fn adv_produce_blocks_on(
+        &mut self,
+        num_blocks: BlockHeight,
+        only_valid: bool,
+        block_height: Option<BlockHeight>,
+        prev_block_height: Option<BlockHeight>,
```

Comment:
Not sure how, but it feels like we can one option or a `BlockProductionMode` enum.
If I just look at API of `adv_produce_blocks_on`, it's confusing how it can happen that block height exists prev block height does not. Or vice versa. Maybe the naming should clarify the purpose more.

Reply:
I improved (at least that was the intention) the API of `adv_produce_blocks_on` by introducing an enum. There are quite a few cases, more than we need, but I went for completeness.